src/transformers/models/poolformer/modeling_poolformer.py

```diff
@@ -0,0 +1,621 @@
+# coding=utf-8
+# Copyright 2021 Google AI, Ross Wightman, The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch PoolFormer model. """
+
+
+import collections.abc
+import math
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+import torch
+from torch.functional import block_diag, norm
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import CrossEntropyLoss, MSELoss
+
+from ...activations import ACT2FN
+from ...file_utils import ModelOutput, add_start_docstrings, add_start_docstrings_to_model_forward, replace_return_docstrings
+from ...modeling_outputs import BaseModelOutput, BaseModelOutputWithPooling, SequenceClassifierOutput
+from ...modeling_utils import PreTrainedModel
+from ...utils import logging
+from .configuration_poolformer import PoolFormerConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "seaailabs/poolformer_s12"
+_CONFIG_FOR_DOC = "PoolFormerConfig"
+
+POOLFORMER_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "seaailabs/poolformer_s12",
+    # See all PoolFormer models at https://huggingface.co/models?filter=poolformer
+]
+
+
+# Inspired by
+# https://github.com/rwightman/pytorch-image-models/blob/b9bd960a032c75ca6b808ddeed76bee5f3ed4972/timm/models/layers/helpers.py
+# From PyTorch internals
```

Comment:
```suggestion
# Copied from transformers.models.vit.modeling_vit.to_2tuple
```
Everywhere where code is identical to an existing piece of code in another modeling file, you can add "copied from" statements. When running "make fix-copies" locally, it will then make sure the copies are consistent.